domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/genericlayout/GenericLayoutEntityValidatorImpl.java

```diff
@@ -34,6 +39,34 @@ public ValidationResult validate(GenericLayoutEntity entity) {
                         (entity.getCaseType() != null ? entity.getCaseType().getReference() : "")
                     ), entity)
             );
+        } else {
+            if (entity.getCaseFieldElementPath() != null) {
+                if (entity.getCaseField().isComplexFieldType() || entity.getCaseField().isCollectionFieldType()) {
+                    List<CaseFieldEntity> caseFields = entity.getCaseType().getCaseFields();
+
+                    List<String> allPaths = CaseFieldEntityUtil.buildDottedComplexFieldPossibilities(caseFields);
```

Comment:
is there a potential performance issue here? We are building all the paths multiple times for a same case type

Reply:
No performance issue here. It takes from 1 to 2 ms in average for one execution if ListElementCode is present. Additionally our definition import happens rarely.